src/test/resources/io/jenkins/plugins/gitlabserver/casc/configuration-as-code.yml

```diff
@@ -0,0 +1,16 @@
+credentials:
+  system:
+    domainCredentials:
+      - credentials:
+          - gitlabPersonalAccessToken:
+              scope: SYSTEM
+              id: "i<3GitLab"
+              token: "XfsqZvVtAx5YCph5bq3r"
+
+unclassified:
+  gitLabServers:
+    servers:
+      - credentialsId: "i<3GitLab"
+        manageHooks: true
+        name: "gitlab.com"
```

Comment:
As gitlab name is randomly generated, is there a way tonot populate the name field?

Reply:
Feel free to test it locally? :sweat_smile:

Reply:
```
unclassified:
  gitLabServers:
    servers:
      - credentialsId: "i<3GitLab"
        manageHooks: true
        serverUrl: "https://gitlab.com"
```

This works. But the randomly generated name cannot be checked in the test. Need to use some different ways to compare them. Like using a regex for name field and comparing each line etc.

Reply:
contains("gitlab") would do?

Reply:
I guess so. My version of solving it:

Test
```java
 @Test
    public void should_support_configuration_export() throws Exception {
        ConfiguratorRegistry registry = ConfiguratorRegistry.get();
        ConfigurationContext context = new ConfigurationContext(registry);
        CNode gitLabServers = getUnclassifiedRoot(context).get("gitLabServers");

        BufferedReader exported = new BufferedReader(new StringReader(toYamlString(gitLabServers)));

        BufferedReader expected = new BufferedReader(new StringReader(toStringFromYamlFile(this, "expected_output.yml")));

        assertThat(exported.readLine(), is(expected.readLine()));
        assertThat(exported.readLine(), is(expected.readLine()));
        assertThat(exported.readLine(), is(expected.readLine()));
        assertThat(exported.readLine(), matchesPattern(expected.readLine()));
        assertThat(exported.readLine(), is(expected.readLine()));
    }
```
Expected output yaml:

```java
servers:
- credentialsId: "i<3GitLab"
  manageHooks: true
  name: "gitlab-[0-9][0-9][0-9][0-9]"
  serverUrl: "https://gitlab.com"
```

Reply:
This works for String as well:
```
String exported = toYamlString(gitLabServers);

String expected = toStringFromYamlFile(this, "expected_output.yml");

assertThat(exported, matchesPattern(expected));
```